ddapm_test_agent/agent.py

```diff
@@ -357,26 +378,52 @@ async def _handle_traces(self, request: Request, version: Literal["v0.4", "v0.5"
                     log.info("Chunk %d could not be displayed (might be incomplete).", i)
             log.info("end of payload %s", "-" * 40)
 
-            with CheckTrace.add_frame(f"payload ({len(traces)} traces)"):
-                await checks.check(
-                    "trace_count_header",
-                    headers=request.headers,
-                    num_traces=len(traces),
-                )
+            if not disable_trace_count_header:
+                with CheckTrace.add_frame(f"payload ({len(traces)} traces)"):
+                    await checks.check(
+                        "trace_count_header",
+                        headers=headers,
+                        num_traces=len(traces),
+                    )
 
         agent_url = request.app["agent_url"]
-        if agent_url:
+        if agent_url and proxy_to_agent:
```

Comment:
let's move this proxying up so that we proxy right away to avoid having to handle errors in the test agent to get to this point